src/boot.s

```diff
@@ -261,7 +261,7 @@ exception_dispatch:
 
 syscall_dispatch:                       # a7 contains syscall index
         la      t0, syscall_vector
-        li      t1, 5                   # TODO: 5 is the number of entries in syscall_vector. We need to find a way to
+        li      t1, 21                  # TODO: 21 is the number of entries in syscall_vector. We need to find a way to
```

Comment:
I am thinking to add:
uint32_t __attribute__((__section__(".text#"))) syscall_vector_len =  ARRAY_SIZE(syscall_vector);

Otherwise, I can't figure a way to use static sizeof() like construct on a variable defined in C.

Reply:
I was thinking to do some code generation, a-la writing a trivial .inc file on every build, but haven't investigated the possibilities yet.